ide/o.n.swing.dirchooser/src/org/netbeans/swing/dirchooser/DirectoryChooserUI.java

```diff
@@ -1911,8 +1871,7 @@ private void addItem(File directory) {
             }
             
             // create File instances of each directory leading up to the top
-            File sf = useShellFolder? getShellFolderForFile(canonical) : canonical;
-            File f = sf;
+            File f = canonical;
```

Comment:
this didn't seem to make a difference, removing this call allows us to remove the `ShellFolder.getShellFolder(file)` call site entirely, which has no public API equivalent.

There is a chance that this is the place which injected the broken `java.io.File` instances. If the public `FileSystemView` methods would have cause this, other projects would likely have noticed it too by now.